### PR TITLE
Fix incompatible type for function return var (Fix #304)

### LINKED ISSUE #304: module mod1 integer, parameter :: fld_size = 10    contains   real(kind=8) function fld_fr()    dimension :: fld_fr(fld_size)    end function fld_fr  end module mod1
The following code fails with the error:
```bash
"../test/testdata/issue304.f90:mod1", line 7: incompatible type declarations, fld_fr
```

```fortran 
module mod1
integer, parameter :: fld_size = 10
contains

 real(kind=8) function fld_fr()
   dimension :: fld_fr(fld_size)
 end function fld_fr

end module mod1
```

### PR DESCRIPTION
**Pull request by clementval**
Fix for #304

### CODE REVIEW

**Comment by h-murai** [comment]
@clementval If merging this PR, an error in the following code isn't detected.

```fortran
real function f()
  integer, dimension(10) :: f
end function f
```

**Comment by clementval** [comment]
@h-murai Ok, I'll look into that.

**Comment by clementval** [comment]
@h-murai Sorry, I didn't have time to look at this recently but I'll try to fix this by the end of this week.

**Comment by clementval** [comment]
@h-murai Would this be better?

File: F-FrontEnd/src/F-compile-decl.c [hunk 1]
```diff
@@ -2301,11 +2301,15 @@ declare_id_type(ID id, TYPE_DESC tp)
             tpp = tp;
             while(TYPE_REF(tpp) != NULL && IS_ARRAY_TYPE(TYPE_REF(tpp)))
                 tpp = TYPE_REF(tpp);
-            if (TYPE_REF(tpp) != NULL && !type_is_soft_compatible(tq, TYPE_REF(tpp)))
-                goto no_compatible;
-            if (TYPE_REF(tpp) == NULL ||
```
Comment (h-murai) [inline F-FrontEnd/src/F-compile-decl.c]: Why do you remove this code (lines 2306-2307) ?

Comment (clementval) [reply F-FrontEnd/src/F-compile-decl.c]: because they are replaced by the one below

**Comment by clementval** [comment]
Migrated to omni-compiler/xcodeml-tools#9

**Pull request closed without merging**